### PR TITLE
interpolation of wipe melt animation, adapted from R&R

### PR DESCRIPTION
**Pull request by rfomin**
The speed of the game is 30% so that the effect is clearly visible:

https://github.com/user-attachments/assets/3f963fa3-c0fb-4ea5-856d-4a6514a5d305

### CODE REVIEW

File: src/d_main.c [hunk 1]
```diff
@@ -432,29 +432,24 @@ void D_Display (void)
...
-          I_Sleep(1);
-          nowtime = I_GetTime();
-          tics = nowtime - wipestart;
-        }
-      while (!tics);
-      wipestart = nowtime;
+      int nowtime = I_GetTime();
+      int tics = nowtime - wipestart;
+
+      fractionaltic = I_GetFracTime();
```
Comment (ceski-1) [inline src/d_main.c:444]: Are there any side effects to updating `fractionaltic` more than once per frame (usually done only at the top of `D_Display()`)? Would it make more sense or be safer to update a separate variable local to `f_wipe.c`?

Also, any concerns about demo sync?

File: src/f_wipe.c [hunk 2]
```diff
@@ -102,114 +91,177 @@ static int wipe_exit(int width, int height, int ticks)
...
+    V_PutBlock(0, 0, width, height, wipe_scr_end);
+    V_RestoreBuffer();
+
+    for (int col = 0; col < wipe_columns; ++col)
     {
-      *d = *s;
-      d += video.pitch;
-      s++;
+        int current = FixedToInt(
+            LerpFixed(IntToFixed(prevy[col]), IntToFixed(curry[col])));
```
Comment (ceski-1) [inline src/f_wipe.c:193]: Whenever interpolation is used, like when calling `LerpFixed()`, there's always an `uncapped` check with a different code path to handle capped framerate. Currently, when using capped fps, the wipe still updates at a higher rate. Compare it to a manually set fps limit of 35 which actually updates at the correct rate.

**Review by ceski-1 — COMMENTED** [review]
Very smooth!

File: src/d_main.c (hunk 1 above)
Comment (rfomin) [reply src/d_main.c:444]: > Are there any side effects to updating `fractionaltic` more than once per frame (usually done only at the top of `D_Display()`)?

For wipe, we only update it in a local loop, once per frame. Overall, I think it's safe, there's just no reason to update it more often than once per frame.

 > Also, any concerns about demo sync?

As long as we spend the same amount of `gametic`s as before, it should be fine.

File: src/f_wipe.c (hunk 2 above)
Comment (rfomin) [reply src/f_wipe.c]: Should be fixed here https://github.com/fabiangreffrath/woof/pull/1903/commits/b9ed42c0790bdf53684f2342903627511ab8af6c

Comment (ceski-1) [reply src/f_wipe.c]: Works well now, thanks.

**Review by ceski-1 — APPROVED** [review]

**Pull request merged**